druid/src/widget/button.rs

```diff
@@ -13,146 +13,98 @@
 // limitations under the License.
 
 //! A button widget.
-
 use crate::theme;
-use crate::widget::{Label, LabelText};
+use crate::widget::{Flex, Label, LabelText, MainAxisAlignment, Painter};
 use crate::{
-    Affine, BoxConstraints, Data, Env, Event, EventCtx, Insets, LayoutCtx, LifeCycle, LifeCycleCtx,
-    LinearGradient, PaintCtx, Point, Rect, RenderContext, Size, UnitPoint, UpdateCtx, Widget,
+    Data, Insets, LinearGradient, Point, Rect, RenderContext, UnitPoint, Widget, WidgetExt,
 };
 
 // the minimum padding added to a button.
 // NOTE: these values are chosen to match the existing look of TextBox; these
 // should be reevaluated at some point.
-const LABEL_INSETS: Insets = Insets::uniform_xy(8., 2.);
+const LABEL_INSETS: Insets = Insets::uniform_xy(8., 4.);
 
 /// A button with a text label.
 pub struct Button<T> {
-    label: Label<T>,
-    label_size: Size,
-    /// A closure that will be invoked when the button is clicked.
-    action: Box<dyn Fn(&mut EventCtx, &mut T, &Env)>,
+    phantom: std::marker::PhantomData<T>,
```

Comment:
I don't think we need this? It doesn't seem like button needs any generic params if we're just sort of using it as a namespace